kernel/kernel-api/src/main/java/io/delta/kernel/data/VectorComparator.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.data;
+
+import io.delta.kernel.annotation.Evolving;
+
+/**
+ * Comparator interface for lazy evaluating comparator
+ */
+@Evolving
+public interface VectorComparator {
```

Comment:
This is applicable only for the default table client module. Please move this to `kernel-defaults` module in package io.delta.kernel.defaults.internal.expressions.

Reply:
thanks for pointing that out, fixed.